app/models/micropost.rb

```diff
@@ -0,0 +1,17 @@
+class Micropost < ApplicationRecord
+  belongs_to :user
+  scope :by_date -> {order created_at: :desc}
```

Comment:
Where you used this scope (?)

Reply:
I don't understand your question but If you mean I used scope for what, So It 's created_at attribute of Micropost table, scope is  class method sort micropost by created_at when object(micropost) is called.

Reply:
Em, when you define a scope, you have to call this scope explicitly. Hence my question is from where you called this scope ?

Reply:
I called scope from show method of user controller

Reply:
@maiq721 
```ruby 
def show
  @microposts = @user.microposts.page(params[:page]).per Settings.user_controller.show_5
end
```
1. I don't see any scope you called here. 
2. You are retrieving just one record here. So no need to call you defined scope. 
3. You can call this scope after you create a micropost, because after creating a micropost, you are redirecting to the root path and there you need to show all of the latest micropost in the descending order. 
4. Please follow the comment in the microposts controller create action. 
5. Scope returns a collection of objects. It would be fine if you study about it more.